packages/stream-model-instance/src/model-instance-document.ts

```diff
@@ -279,38 +196,48 @@ export class ModelInstanceDocument<T = Record<string, any>> extends Stream {
    * @param content - genesis content
    * @param metadata - genesis metadata
    */
-  static async makeGenesis<T>(
+  private static async _makeGenesis<T>(
     signer: CeramicSigner,
-    content: T | null | undefined,
-    metadata?: ModelInstanceDocumentMetadataArgs
+    content: T | null,
+    metadata: ModelInstanceDocumentMetadata
   ): Promise<CeramicCommit> {
-    if (!metadata) {
-      metadata = {}
+    if (!metadata.model) {
+      throw new Error(`Must specify a 'model' when creating a ModelInstanceDocument`)
     }
 
-    if (!metadata.controllers || metadata.controllers.length === 0) {
+    if (!metadata.controller) {
       if (signer.did) {
         await _ensureAuthenticated(signer)
-        // When did has parent, it has a capability, the did issuer (parent) of the capability is the stream controller
-        metadata.controllers = [signer.did.hasParent ? signer.did.parent : signer.did.id]
+        // When did has a parent, it has a capability, and the did issuer (parent) of the capability
+        // is the stream controller
+        metadata.controller = signer.did.hasParent ? signer.did.parent : signer.did.id
       } else {
-        throw new Error('No controllers specified')
+        throw new Error('No controller specified')
       }
     }
-    if (metadata.controllers?.length !== 1) {
-      throw new Error('Exactly one controller must be specified')
-    }
-
-    const header: GenesisHeader = headerFromMetadata(metadata, true)
 
-    if (content == null) {
-      const result = { header }
-      // Check if we can encode it in cbor. Should throw an error when invalid payload.
-      dagCbor.encode(result)
-      // No signature needed if no genesis content
-      return result
+    // TODO(NET-1464): enable GenesisHeader to receive 'controller' field directly
+    const header: GenesisHeader = {
+      controllers: [metadata.controller],
+      unique: uint8arrays.toString(randomBytes(12), 'base64'),
```

Comment:
Maybe we can make this raw bytes as well to save space?
```suggestion
      unique: randomBytes(12),
```
Just a  thought, not super important.